### ISSUE TITLE
False positive test due to assertions around a revoked credential 

### ISSUE THREAD

**Issue by gannan08** [w3c-ccg/vc-http-api#132]
The test suite currently uses the following [Revocation List](https://github.com/w3c-ccg/vc-http-api/blob/master/docs/fixtures/revocationList.json). Its entry at index 1 is set as revoked.

The following credential found [here](https://github.com/w3c-ccg/vc-http-api/blob/4461a9ed7d788d243941482442439d6b7d6e0020/packages/vc-http-api-test-server/__fixtures__/verifiableCredentials/case-15.json) has its `revocationListIndex` set to 1, therefore it will always fail verification for vendors that handle RevocationList2020.

We are failing a [test](https://github.com/w3c-ccg/vc-http-api/blob/4461a9ed7d788d243941482442439d6b7d6e0020/packages/vc-http-api-test-server/__tests__/verifyCredential.spec.js#L268) due to an assertion for a `200` OK Response Status Code from an HTTP Server. We believe our verifier is correct in returning the `400` HTTP Status Code as the credential is revoked.

I propose disabling [case-15](https://github.com/w3c-ccg/vc-http-api/blob/4461a9ed7d788d243941482442439d6b7d6e0020/packages/vc-http-api-test-server/__fixtures__/verifiableCredentials/case-15.json) or using a credential that is not revoked. If that is not amicable, I'm open to other options.

P.S. If the behavior is to report a revoked credential as successfully verified, then I may have missed the spec language around that. Additionally, I will go ahead and close the issue if that's the case.

```js
{
    '@context': [
      'https://www.w3.org/2018/credentials/v1',
      'https://w3id.org/citizenship/v1',
      'https://w3id.org/vc-revocation-list-2020/v1'
    ],
    id: 'https://issuer.oidp.uscis.gov/credentials/83627465',
    type: [ 'VerifiableCredential', 'PermanentResidentCard' ],
    identifier: '83627465',
    name: 'Permanent Resident Card',
    description: 'Government of Example Permanent Resident Card.',
    issuanceDate: '2019-12-03T12:19:52Z',
    expirationDate: '2029-12-03T12:19:52Z',
    credentialStatus: {
      id: 'https://issuer.oidp.uscis.gov/credentials/status/3#94567',
      type: 'RevocationList2020Status',
      revocationListIndex: '1',
      revocationListCredential: 'https://w3c-ccg.github.io/vc-http-api/fixtures/revocationList.json'
    },
    credentialSubject: {
      id: 'did:example:b34ca6cd37bbf23',
      type: [Array],
      givenName: 'JOHN',
      familyName: 'SMITH',
      gender: 'Male',
      image: 'data:image/png;base64,iVBORw0KGgo...kJggg==',
      residentSince: '2015-01-01',
      lprCategory: 'C09',
      lprNumber: '999-999-999',
      commuterClassification: 'C1',
      birthCountry: 'Bahamas',
      birthDate: '1958-07-17'
    },
    issuer: 'did:key:z6MkiY62766b1LJkExWMsM3QG4WtX7QpY823dxoYzr9qZvJ3',
    proof: {
      type: 'Ed25519Signature2018',
      created: '2021-02-18T04:19:23Z',
      jws: 'eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..tM_94zQxF5LwVqaW1Q4iGnkEDyn9Mbg7A6l3d1sVd9-Yt6AOVvAILqYKWuGiXIOy64Eudw_y0UMJ4I0XpEeUAw',
      proofPurpose: 'assertionMethod',
      verificationMethod: 'did:key:z6MkiY62766b1LJkExWMsM3QG4WtX7QpY823dxoYzr9qZvJ3#z6MkiY62766b1LJkExWMsM3QG4WtX7QpY823dxoYzr9qZvJ3'
    }
  }
```

cc: @tplooker @OR13

**Comment by OR13** [comment]
I would assert that:

```
POST /verify { presentation containing revoked credential } => 200 ( verified: false, reasons: [] )
```

A 400 error is not the correct response code if the client as constructed a well formed request.

**Comment by tplooker** [comment]
Ganesh, the test case is to check for a 400 when the credential is revoked? [See here](https://github.com/w3c-ccg/vc-http-api/blob/master/packages/vc-http-api-test-server/__tests__/verifyCredential.spec.js#L229)

**Comment by tplooker** [comment]
Therefore case 15 should be returning a 400 when attempting to verify the credential status, also agree with @OR13 though that semantically this is not accurate and should be revised at a later date

**Comment by gannan08** [comment]
Let me summarize...

Based on the current specification, the correct response code for a revoked credential is `400`. IOW, to pass the test suite now, return a `400` status code.

In the future, the current specification will change the response code to `200`. I agree. But why? The 4xx Status Codes should be used to convey an exception or an error during execution. If a credential is successfully determined to be revoked, then the API should successfully return the result.  This means one should return a `200` status code with a response that says the credential is revoked.

Another way to explain this... it is improper to throw an exception or an error if a function executes successfully for a given input.

Am I aligned? If so, we can close the issue.

PS - The original problem was on our end. The [test](https://github.com/w3c-ccg/vc-http-api/blob/4461a9ed7d788d243941482442439d6b7d6e0020/packages/vc-http-api-test-server/__tests__/verifyCredential.spec.js#L261-L269) only specifies the `proof` option. Our software additionally verified the credential's status as well, hence the `400` status code for the [case-15](https://github.com/w3c-ccg/vc-http-api/blob/4461a9ed7d788d243941482442439d6b7d6e0020/packages/vc-http-api-test-server/__fixtures__/verifiableCredentials/case-15.json) credential. That has been resolved.
